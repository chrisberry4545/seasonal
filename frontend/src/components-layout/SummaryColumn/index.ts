export * from './SummaryColumn';

export * from './SummaryColumn.interface';
