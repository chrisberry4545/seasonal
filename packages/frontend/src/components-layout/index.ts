export * from './ImageGridItem/ImageGridItem';

export * from './ImageGrid/ImageGrid';

export * from './SummaryColumn';

export * from './SummaryColumnList';

export * from './SeasonNameView/SeasonNameView';

export * from './Toast/Toast';
