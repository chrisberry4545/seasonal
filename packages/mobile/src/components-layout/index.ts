export * from './ConfirmButtonLayout/ConfirmButtonLayout';

export * from './DefaultPaddingContainer/DefaultPaddingContainer';

export * from './DetailsPageLayout/DetailsPageLayout';

export * from './FeedbackQuestionsLayout/FeedbackQuestionsLayout';

export * from './HeaderAndBackButton/HeaderAndBackButton';

export * from './ImageGrid/ImageGrid';

export * from './ImageGridItem/ImageGridItem';

export * from './MainContainer/MainContainer';

export * from './ModalLayout/ModalLayout';

export * from './NoResultsText/NoResultsText';

export * from './SeasonNameView/SeasonNameView';

export * from './SwitchableGridOrList/SwitchableGridOrList';

export * from './Toast/Toast';

export * from './TopLoadingSpinner/TopLoadingSpinner';
