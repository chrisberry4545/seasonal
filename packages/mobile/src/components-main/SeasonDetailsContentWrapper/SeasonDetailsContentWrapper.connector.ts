import {
  ISeasonDetailsContentWrapperInputProps
} from './SeasonDetailsContentWrapper.interface';

import { connect } from 'react-redux';
import {
  SeasonDetailsContentWrapper
} from './SeasonDetailsContentWrapper';
import {
  selectIsBasicSeasonsLoading, selectCurrentSeasonName
} from '@chrisb-dev/seasonal-shared-frontend-redux';
import { IState } from '../../store';

const mapStateToProps = (
  state: IState
): ISeasonDetailsContentWrapperInputProps => {
  return {
    currentSeasonName: selectCurrentSeasonName(state),
    isLoading: selectIsBasicSeasonsLoading(state)
  };
};

export const SeasonDetailsContentWrapperConnector = connect(
  mapStateToProps
)(SeasonDetailsContentWrapper);
