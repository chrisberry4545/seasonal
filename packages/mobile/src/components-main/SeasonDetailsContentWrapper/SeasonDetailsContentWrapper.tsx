import React, { FC, Fragment } from 'react';
import { ScrollView } from 'react-native';
import {
  HeaderConnecter
} from '../Header/Header.connector';

import {
  ISeasonDetailsContentWrapperInputProps
} from './SeasonDetailsContentWrapper.interface';
import {
  MainContainer,
  TopLoadingSpinner,
  SeasonNameView
} from '../../components-layout';

export const SeasonDetailsContentWrapper:
FC<ISeasonDetailsContentWrapperInputProps> = ({
  isLoading,
  currentSeasonName,
  children
}) => (
      <MainContainer>
        <HeaderConnecter />
        <ScrollView>
          <SeasonNameView name={currentSeasonName} />
          {
            !isLoading
              ? <Fragment>{children}</Fragment>
              : <TopLoadingSpinner />
          }
        </ScrollView>
      </MainContainer>
);
