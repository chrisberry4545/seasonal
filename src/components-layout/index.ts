export * from './ImageGrid/ImageGrid';

export * from './ImageGridItem/ImageGridItem';

export * from './MainContainer/MainContainer';

export * from './SeasonNameView/SeasonNameView';
