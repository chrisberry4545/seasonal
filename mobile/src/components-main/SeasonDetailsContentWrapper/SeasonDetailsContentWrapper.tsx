import React, { FC, Fragment } from 'react';
import { ScrollView, ViewStyle, TextStyle } from 'react-native';
import {
  HeaderConnecter
} from '../../components-main/Header/Header.connector';

import {
  ISeasonDetailsContentWrapperInputProps
} from './SeasonDetailsContentWrapper.interface';
import { MainContainer, TopLoadingSpinner } from '../../components-layout';
import { TextHeadingSmall } from '../../components-elements';

const styleSeasonDetailsContentWrapper: TextStyle = {
  height: 20,
  marginBottom: 12,
  marginTop: 12,
  textAlign: 'center',
  width: '100%'
};

const styleSeasonalDetailsContentWrapper: ViewStyle = {
  flex: 1
};

export const SeasonDetailsContentWrapper:
FC<ISeasonDetailsContentWrapperInputProps> = ({
  isLoading,
  currentSeasonName,
  children
}) => (
      <MainContainer>
        <HeaderConnecter />
        <ScrollView style={ styleSeasonalDetailsContentWrapper }>
          <TextHeadingSmall style={styleSeasonDetailsContentWrapper}>
            {currentSeasonName}
          </TextHeadingSmall>
          {
            !isLoading
              ? <Fragment>{children}</Fragment>
              : <TopLoadingSpinner />
          }
        </ScrollView>
      </MainContainer>
);
