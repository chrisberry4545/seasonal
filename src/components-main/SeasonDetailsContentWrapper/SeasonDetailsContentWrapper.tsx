import React, { FC } from 'react';
import { ScrollView, ViewStyle } from 'react-native';
import {
  HeaderConnecter
} from '../../components-main/Header/Header.connector';

import {
  ISeasonDetailsContentWrapperInputProps
} from './SeasonDetailsContentWrapper.interface';
import { MainContainer, TopLoadingSpinner } from '../../components-layout';

const styleSeasonalDetailsContentWrapper: ViewStyle = {
  flex: 1
};

export const SeasonDetailsContentWrapper:
FC<ISeasonDetailsContentWrapperInputProps> = ({
  isLoading,
  children
}) => (
      <MainContainer>
        <HeaderConnecter />
        <ScrollView style={ styleSeasonalDetailsContentWrapper }>
          {
            !isLoading
              ? children
              : <TopLoadingSpinner />
          }
        </ScrollView>
      </MainContainer>
);
