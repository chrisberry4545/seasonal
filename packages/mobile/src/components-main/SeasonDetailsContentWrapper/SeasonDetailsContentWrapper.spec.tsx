import React from 'react';
import { SeasonDetailsContentWrapper } from './SeasonDetailsContentWrapper';
import { shallow, ShallowWrapper } from 'enzyme';
import { TopLoadingSpinner } from '../../components-layout';

jest.mock('../Header/Header.connector', () => ({
  HeaderConnecter: () => 'HeaderConnecter'
}));
jest.mock('../../components-layout', () => ({
  MainContainer: () => 'MainContainer',
  SeasonNameView: () => 'SeasonNameView',
  TopLoadingSpinner: () => 'TopLoadingSpinner'
}));

describe('<SeasonDetailsContentWrapper />', () => {
  let wrapper: ShallowWrapper;

  describe('when it is not loading', () => {
    beforeEach(() =>
      wrapper = shallow(
        <SeasonDetailsContentWrapper
          isLoading={false}
          currentSeasonName='January'>
            Children
        </SeasonDetailsContentWrapper>
      )
    );

    test('renders correctly', () =>
      expect(wrapper).toMatchSnapshot());
  });

  describe('when it is loading', () => {
    beforeEach(() =>
      wrapper = shallow(
        <SeasonDetailsContentWrapper
          isLoading={true}
          currentSeasonName='January'>
            Children
        </SeasonDetailsContentWrapper>
      )
    );

    test('shows a loading spinner', () =>
      expect(wrapper.find(TopLoadingSpinner).exists()).toBe(true));

  });

});
