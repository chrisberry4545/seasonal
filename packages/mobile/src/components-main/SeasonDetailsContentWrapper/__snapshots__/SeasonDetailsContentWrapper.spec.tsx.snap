// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SeasonDetailsContentWrapper /> when it is not loading renders correctly 1`] = `
<MainContainer>
  <HeaderConnecter />
  <ScrollViewMock>
    <SeasonNameView
      name="January"
    />
    Children
  </ScrollViewMock>
</MainContainer>
`;
